openmdao/docs/theory_manual/total_derivs/vectorized.rst

```diff
@@ -0,0 +1,93 @@
+
+While reduced computational overhead is advantageous, this approach also gives an increased memory footprint.
+As we noted, in order to use this algorithm multiple right-hand-side vectors and solutions vectors must be allocated at the same time.
+
+.. tip::
+
+    If you are considering using vectorized total derivatives solves because you have a set of vectorized components in your model, you might also consider checking if your problem has a separable structure that can be taken advantage of with :ref:`simultaneous derivatives<theory_separable_variables>`.
+    If you can use simultaneous derivatives, you get all of the computational speed up, with none of the memory cost.
+    If your problem isn't separable, then you consider vectorized derivatives instead.
+
+-----------------------------
+How Much Memory Does it Use?
+-----------------------------
+
+OpenMDAO allocates vectors for the linear solves according to how many output variables there are and how large each one is.
+How much additional memory is needed will depend on the details of your model.
+OpenMDAO only needs to allocate enough memory for the the vectorized solve of one variable at a time, so the memory cost can be considered on a per-variable basis and will be governed by whichever variable has the largest size.
+
+Consider a notional model (or a part of a model) with 5 design variables and 5 constraints, each one of size 1000.
+This notional model is very simple, comprised of just two components: 1 :ref:`IndepVarComp<comp-type-1-indepvarcomp>` and one :ref:`ExplicitComponent<comp-type-2-explicitcomp>`.
+So there are 10 total outputs in the model, yielding a vector of size 10000 using 0.08 MB of memory.
+OpenMDAO needs two vectors for each linear solve (one for the solution and one for the right hand side), using 0.16 MB for the pair.
+
+A non-vectorized solve would require just the single pair of vectors,
+but a vectorized derivative solve needs 1000 right hand side vectors and 1000 solution vectors all at once, using 160 MB of memory.
+
+On modern computers 160 MB of memory is not a significant amount, especially if it comes with the benefit of a large computational savings.
+However, the above example only involved 2 components and hence the cost of the recursive solves won't be significant either.
+
+So instead, consider what would happen if you had the same 5 design variables and 5 constraints of size 1000, but now the model is comprised of 15 different components connected in a feed-forward manner arranged in some complex hierarchy.
+Each component will have its own 5 output variables, giving a total of 75 variables of size 1000 each.
+In this larger model each vector is now of size 75000 using 0.6 MB of memory.
+So a non-vectorized solve will use 1.2 MB of memory but the vectorized solve will now use 1200 MB (1.2 GB) of memory.
```

Comment:
Just for sanity, have we verified that the actual increase in memory matches what we expect?

Reply:
I haven't verified it. But this is the theoretical manual :) 

You're right that we should check it out, but its tough to isolate just that one section of memory usage. I don't know how wed do that.